### PR TITLE
Cargo.toml changes needed for publishing on crates.io

### LINKED ISSUE #376: Publish on crates.io
It would be nice to play well with the Rust ecosystem. This is a blocker to publishing anything that depends on mozjs, e.g. https://github.com/asajeffrey/josephine/issues/35

cc: @jdm @tschneidereit @fitzgen @nox

### PR DESCRIPTION
**Pull request by fitzgen**
r? @asajeffrey

Fixes #376 

---
This change is [Reviewable](https://reviewable.io/reviews/servo/rust-mozjs/377)

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -34,11 +36,9 @@ doctest = false
...
 
-[dependencies.mozjs_sys]
-git = "https://github.com/servo/mozjs"
-
 [dependencies]
 lazy_static = "0.2.1"
 libc = "0.2"
 log = "0.3"
 num-traits = "0.1.32"
+mozjs_sys = "0.0.0"
```
Comment (KiChjang) [inline Cargo.toml]: The mozjs_sys version on crates.io is 0.1.0 instead of 0.0.0. https://crates.io/crates/mozjs_sys

Comment (fitzgen) [reply Cargo.toml]: Huh. I didn't realize it had ever been published before.

Comment (fitzgen) [reply Cargo.toml]: @SimonSapin, can you add myself and others (jdm, nox, asajeffrey, KiChjang, etc?) as owners? Thanks!

Comment (SimonSapin) [reply Cargo.toml]: @fitzgen I’ve added you. The `servo/cargo-publish` team already has access.

Comment (SimonSapin) [reply Cargo.toml]: Since we’re moving things around anyway we could pick one of `mozjs` vs `js` vs something else for both crates?

Comment (fitzgen) [reply Cargo.toml]: D'oh -- I meant to rename this crate to `mozjs` since `js` is already taken on crates.io. Good eye!

**Comment by fitzgen** [comment]
I didn't change the version for this crate -- did we want to do that before publishing?

**Comment by asajeffrey** [comment]
I guess we can't keep the version in sync with the SM version, unless we want to limit ourselves to only releasing breaking changes when we change SM. We may as well leave the version alone.

**Comment by asajeffrey** [comment]
Once the crate is published, we can merge https://github.com/servo/servo/pull/19138

**Pull request merged**